cmd/ctr/run.go

```diff
@@ -103,8 +111,11 @@ var runCommand = cli.Command{
 			id          = context.Args().Get(1)
 			imageRef    = context.Args().First()
 			tty         = context.Bool("tty")
+			detach      = context.Bool("detach")
 		)
-		defer cancel()
+		if !detach {
+			defer cancel()
```

Comment:
What is the effect of calling cancel here? Is there something in the client code that picks this up and cleans up? I am assuming that there are not ongoing rpc requests which this would have an effect on.

Reply:
you are right, let me change this